drivers/syscon/efuse_bflb.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT bflb_efuse
+
+#include <zephyr/drivers/syscon.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(efuse_bflb, CONFIG_SYSCON_LOG_LEVEL);
+
+#include <bouffalolab/bl60x/bflb_soc.h>
+#include <bouffalolab/bl60x/hbn_reg.h>
+#include <bouffalolab/bl60x/ef_ctrl_reg.h>
+#include <bouffalolab/bl60x/extra_defines.h>
+
+#define EF_CTRL_EFUSE_CYCLE_PROTECT (0xbf << 24)
+#define EF_CTRL_EFUSE_CTRL_PROTECT  (0xbf << 8)
+#define EF_CTRL_OP_MODE_AUTO        0
+#define EF_CTRL_PARA_DFT            0
+
+struct efuse_bflb_data {
+	uint8_t cache[DT_INST_PROP(0, size)];
+	bool cached;
+};
+
+struct efuse_bflb_config {
+	uint32_t addr;
+	size_t size;
+};
+
+
+static void system_clock_settle(void)
```

Comment:
all internal functions should have driver prefix.
consider to add inline when applicable

Reply:
done